src/configuration-store/configuration-store.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * ConfigurationStore interface
+ *
+ * The interface guides implementation
+ * of a policy for handling a mixture of async and sync storage.
+ *
+ * The goal is to support remote API responses to be written to the sync and async store,
+ * while also supporting reading from the sync store to maintain public SDK APIs.
+ *
+ * Implementation is handled in upstream libraries to best support their use case, some ideas:
+ *
+ * - Javascript frontend:
+ *   - SyncStore: backed by localStorage
+ *   - AsyncStore: backed by IndexedDB or chrome.storage.local
+ *
+ * - NodeJS backend:
+ *   - SyncStore: backed by LRU cache
+ *   - AsyncStore: none
```

Comment:
excellent comment! 👏

Reply:
Should SyncStore use Map/Record instead of LRU cache similar to the change we made to Python?